client/app/components/parameters.js

```diff
@@ -77,21 +34,26 @@ function ParametersDirective($location, $uibModal) {
         );
       }
 
-      scope.showParameterSettings = (param) => {
-        $uibModal.open({
-          component: 'parameterSettings',
-          resolve: {
-            parameter: param,
-          },
-        });
+      scope.showParameterSettings = (parameter, index) => {
+        EditParameterSettingsDialog
+          .showModal({ parameter })
+          .result.then((updated) => {
+            // update parameters
+            scope.parameters[index] = extend(parameter, updated);
```

Comment:
Why `extend` and not just override? Extending has the issue of leaving irrelevant settings from previous parameter type.

Reply:
Hmm.. cause of this
```js
const [param, setParam] = useState(clone(props.parameter));`
```
`param` doesn't then hold the non-enumerable "urlParams" and it gets lost...

I'll try to fix this.

Reply:
Actually, it's not the clone. It's this the object spreading that loses it.
```js
setParm({ ...param, prop: newValue });
```

Reply:
@kravets-levko got any good idea to overcome this?

Reply:
> leaving irrelevant settings from previous parameter type

@arikfr That's happening either way since we don't reset irrelevant params anyhow.

Reply:
You're right :-(

Reply:
@ranbena You need to wrap raw object with `Parameter` constructor
https://github.com/getredash/redash/blob/83668a6840075eb979c9cb0aeae9a680153416ee/client/app/services/query.js#L51-L53
and then manually update `urlPrefix` prop using "old" parameter object. That's the only way to do it properly (`clone` will probably do it, except of updating `urlPrefix`; `extend`/spread operator will lose object type).

Reply:
I say let's leave it as is - at least this way the component stays oblivious to `urlPrefix`.

Reply:
👍 let's just open an issue.